src/App.tsx

```diff
@@ -1,36 +1,89 @@
-import { FC, MouseEventHandler, useState } from 'react';
+import { FC, useState } from 'react';
+import { Routes, Route } from 'react-router-dom';
+
+import {
+	Courses,
+	Registration,
+	Login,
+	CreateCourse,
+	Home,
+	CourseInfo,
+	NotFound,
+} from './components';
+import { Layout } from './common';
 
-import { Header, Courses, CreateCourse } from './components';
 import { mockedCoursesList, mockedAuthorsList } from './constants';
-import { MainWrapper } from './common';
 
 const App: FC = () => {
-	const [addingMode, setAddingMode] = useState<boolean>(false);
 	const [courses, setCourses] = useState<Course[]>(mockedCoursesList);
 	const [authors, setAuthors] = useState<Author[]>(mockedAuthorsList);
-	const handleAddNewCourseButtonClick: MouseEventHandler<
-		HTMLButtonElement
-	> = (): void => setAddingMode(true);
+	const [token, setToken] = useState<string | null>(
+		localStorage.getItem('coursesAppUserToken')
+	);
+	const [name, setName] = useState<string | null>(
+		localStorage.getItem('coursesAppUserName')
+	);
+
 	return (
 		<>
-			<Header />
-			<MainWrapper forwardedAs='main' addBorder gap='1rem'>
-				{addingMode ? (
-					<CreateCourse
-						authors={authors}
-						setAuthors={setAuthors}
-						courses={courses}
-						setCourses={setCourses}
-						setAddingMode={setAddingMode}
+			<Routes>
+				<Route
+					path='/'
+					element={<Layout name={name} setName={setName} setToken={setToken} />}
+				>
+					<Route index element={<Home token={token} />} />
+					<Route
+						path='registration'
+						element={token ? <Home token={token} /> : <Registration />}
+					/>
+					<Route
+						path='login'
+						element={
+							token ? (
+								<Home token={token} />
+							) : (
+								<Login token={token} setName={setName} setToken={setToken} />
+							)
```

Comment:
Don't really like your approach with `Home` component.
You just should add a check inside your pages that redirects user where needed

Reply:
For example:
```
const Component = () => {
  const navigate = useNavigate()
  const token = useSelector(tokenSelector)
  
  useEffect(() => {
    !token && navigate('/login')
  }, [ token ])
}
```